### PR TITLE
Fix MergeTask Bug

### PR DESCRIPTION
**Pull request by bgklein**
PR for merge task bug

Will submit PR tommorow for Base SDK error bug

### CODE REVIEW

**Comment by xingwu1** [comment]
@bgklein The code looks good. Could you update the doc about mergeTask?

File: azext/batch/_template_utils.py [hunk 1]
```diff
@@ -994,6 +999,15 @@ def expand_task_factory(job, fileutils):
...
 
+def has_merge_task(job):
+    """ Check if user has specified a mergeTask on the task factory
+    :param job_obj: The JSON job entity loaded from a template.:
+    :return: true if merge task present
+    """
+    if job.task_factory.type in ['parametricSweep', 'taskPerFile'] and job.task_factory.merge_task:
+        return True
+    return False
+
```
Comment (xingwu1) [inline azext/batch/_template_utils.py]: miss one white line?

File: tests/data/batch.job.mergetask.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            "taskFactory": {
+                "type": "taskPerFile",
+                "source" : {
+                    "fileGroup" : "[parameters('testData')]"
+                },
+                "repeatTask": {
+                    "commandLine": "/bin/bash -c 'cat {fileName}'",
+                    "resourceFiles": [
+                        {
+                            "blobSource" : "{url}",
```
Comment (xingwu1) [inline tests/data/batch.job.mergetask.json]: change to httpUrl

File: tests/data/batch.job.mergetask.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                            }
+                        }
+                    ]
+                },
+                "mergeTask" : {
+                    "displayName": "myMergeTask",
+                    "commandLine": "/bin/bash -c 'ls'",
+                    "resourceFiles": [
+                        {
+                            "autoStorageContainerName": "fgrp-output"
```
Comment (xingwu1) [inline tests/data/batch.job.mergetask.json]: is there any way we can get the "fgrp-xxxx" from [testData]?

Comment (bgklein) [reply tests/data/batch.job.mergetask.json]: Yea, we can either add another parameter/variable or specify it as fgrp-[parameters(outputData)

**Pull request merged**